Sources/MapboxMaps/Foundation/Camera/CameraTransition.swift

```diff
@@ -0,0 +1,113 @@
+import UIKit
+import CoreLocation
+
+/// Structure used to represent a desired change to the map's camera
+public struct CameraTransition {
+
+    /// Represents a change to the center coordinate of the map.
+    /// NOTE: Setting the `toValue` of `center` overrides any `anchor` animations
+    public var center: Change<CLLocationCoordinate2D>
+
+    /// Represents a change to the zoom of the map.
+    public var zoom: Change<CGFloat>
+
+    /// Represetns a change to the padding of the map.
+    public var padding: Change<UIEdgeInsets>
+
+    /// Represents a change to the anchor of the map
+    /// NOTE: Incompatible with concurrent center animations
+    public var anchor: Change<CGPoint>
+
+    /// Set this value in order to make bearing/zoom animations
+    /// honor a constant anchor throughout the transition
+    /// NOTE: Incompatible with concurrent center animations
+    public var constantAnchor: CGPoint?
+
+    /// Represents a change to the bearing of the map.
+    public var bearing: Change<Double>
+
+    /// Ensures that bearing transitions are optimized to take the shortest path.
+    public var shouldOptimizeBearingPath: Bool = true
+
+    /// Represents a change to the pitch of the map.
+    public var pitch: Change<CGFloat>
+
+    /// Generic struct used to represent a change in a value from a starting point (i.e. `fromValue`) to an end point (i.e. `toValue`).
+    public struct Change<T> {
+        public var fromValue: T
+        public var toValue: T?
+
+        init(fromValue: T, toValue: T? = nil) {
+            self.fromValue = fromValue
+            self.toValue = toValue
```

Comment:
Does the word `Value` add any information? Would it be just as clear if we omitted it?

Reply:
I'm on the fence about this.

Reply:
I like the `Value` suffix..